src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CcdNewCaseCreator.java

```diff
@@ -183,24 +192,58 @@ private long createNewCaseInCcd(
     private CaseDataContent getCaseDataContent(
         Map<String, Object> caseData,
         String exceptionRecordId,
-        String eventId,
-        String eventToken
+        String envelopeId,
+        StartEventResponse startEventResponse,
+        String service,
+        String loggingContext
     ) {
-        Map<String, Object> data = new HashMap<>(caseData);
-        data.put(EXCEPTION_RECORD_REFERENCE, exceptionRecordId);
+        Map<String, Object> completeCaseData =
+            setBulkScanSpecificFieldsInCaseData(caseData, service, exceptionRecordId, envelopeId, loggingContext);
 
         return CaseDataContent
             .builder()
             .caseReference(exceptionRecordId)
-            .data(data)
+            .data(completeCaseData)
             .event(Event
                 .builder()
-                .id(eventId)
+                .id(startEventResponse.getEventId())
                 .summary("Case created")
                 .description("Case created from exception record ref " + exceptionRecordId)
                 .build()
             )
-            .eventToken(eventToken)
+            .eventToken(startEventResponse.getToken())
             .build();
     }
+
+    private Map<String, Object> setBulkScanSpecificFieldsInCaseData(
+        Map<String, Object> caseData,
+        String service,
+        String exceptionRecordId,
+        String envelopeId,
+        String loggingContext
+    ) {
+        Map<String, Object> updatedCaseData = new HashMap<>(caseData);
+        updatedCaseData.put(BULK_SCAN_CASE_REFERENCE, exceptionRecordId);
+
+        if (envelopeReferenceCollectionHelper.serviceSupportsEnvelopeReferences(service)) {
+            updatedCaseData.put(
+                BULK_SCAN_ENVELOPES,
+                envelopeReferenceCollectionHelper.singleEnvelopeReferenceList(envelopeId, CaseAction.CREATE)
+            );
+
+            log.info("{} field was set in new case data for service service {}",
+                BULK_SCAN_ENVELOPES,
+                service,
+                loggingContext
+            );
+        } else {
+            log.info("{} field was not set in new case data for service service {} - not supported by this service",
```

Comment:
Doubled 'service'. 
I'd consider making 'not' uppercase in the first sentence, for clarity.

Reply:
done